docs/source/data_extraction.rst

```diff
@@ -287,3 +289,43 @@ There are several ways to define columns, depending on the type of the attribute
 * Collections of values
 
  * :class:`~citrine.gemtables.columns.ConcatColumn`: for concatenating the results of a list- or set-valued result, such as is returned by :class:`~citrine.gemtables.variables.IngredientLabelsSetInOutput`
+
+Compatibility with AI Engine
+----------------------------
+
+The Citrine Platform automatically converts the values found in a GEM Table
+into the format used by the :ref:`AI Engine <workflows>` for predictor training and default asset creation.
+This includes generating :ref:`descriptors <descriptors>` from the variables found in the table configuration
+and extracting individual values from the cells of the GEM Table.
+
+In most cases, descriptors are generated based on the bounds
+(children of the :class:`~gemd.entity.bounds.base_bounds.BaseBounds` class)
+found on the attribute template referenced by a :ref:`GEM Table variable <variables>`.
+The key of the descriptor is derived from concatenation of the `headers` field of the table variable.
+An exception to this is for the :class:`~citrine.informatics.descriptors.FormulationDescriptor`
+which follows the special rule described below.
+The mappings from variables in a GEM Table to descriptors are as follows:
+
+- :class:`~gemd.entity.bounds.RealBounds` are converted to a :class:`~citrine.informatics.descriptors.RealDescriptor`
+- :class:`~gemd.entity.bounds.IntegerBounds` are converted to an :class:`~citrine.informatics.descriptors.IntegerDescriptor`
+- :class:`~gemd.entity.bounds.CategoricalBounds` are converted to a :class:`~citrine.informatics.descriptors.CategoricalDescriptor`
+- :class:`~gemd.entity.bounds.CompositionBounds` are converted to a :class:`~citrine.informatics.descriptors.ChemicalFormulaDescriptor`
```

Comment:
if the following statement is true, we should at it to the end of this line:
`or CategoricalDescriptor when the quantity keys are not atomic symbols`

Reply:
That is not how it works, though. A `CompositionBounds` always produces a `ChemicalFormulaDescriptor`.

In order for a `NominalComposition` to work with a `CategoricalDescriptor` in the AIE, you'd have to pair it with a `CategoriclaBounds`. Is that even possible?

Reply:
It looks like a `NominalComposition` actually cant be used with a `CategoricalBounds`, meaning these mappings won't work in practice. Looks like I'll have to remove the option to treat a `NominalComposition` as a categorical in cases where the keys _are not_ valid atomic symbols.

Reply:
I am not sure how it works, but the thing that confuses me about how this works is this part a little further down:
```
- :class:`~citrine.informatics.descriptors.CategoricalDescriptor`: values of type :class:`~gemd.entity.NominalCategorical` and :class:`~gemd.entity.DiscreteCategorical`,
  or values of type :class:`~gemd.entity.NominalComposition` when the quantity keys are **not** atomic symbols
- :class:`~citrine.informatics.descriptors.ChemicalFormulaDescriptor`: values of type :class:`~gemd.entity.EmpiricalFormula`,
  or values of type :class:`~gemd.entity.NominalComposition` when **all** quantity keys are valid atomic symbols
```

It says that NominalComposition can be compatible with CategoricalDescriptor or ChemicalFormulaDescriptor depending on quantity keys being atomic symbols.

Reply:
ok I gotcha 👌

Reply:
Yep, sorry about that. I'll update the docs here too.

The logic will be that a `NominalComposition` can only be used if the keys are all valid atomic symbols.

Reply:
Great thanks for documenting that! Just want to acknowledge, we have had customers want to use a NominalComposition value as a RealDescriptor, but that can be a feature request for the future 🙂

Reply:
> we have had customers want to use a NominalComposition value as a RealDescriptor

How would this be possible? I don't see a clear way in which we treat that value type as a real number?

Reply:
Nominal Compositions are mappings of component to quantity. When a user constructs a gemtable, they can use a `ComponentQuantityColumn` with the component name as the column name, and the quantity of that component for the given attribute will be filled into the cell values

Reply:
I see, that's a bit trickier one, as we'd have to generate a set of real descriptors per value, I guess.